docs/user/adding-new-metrics.md

```diff
@@ -1,97 +1,184 @@
+  
+![Metric lifetime timeline](metric-lifetime-timeline.svg)
+
+- **A. Ping lifetime, set on change**: The value isn't included in Ping 1, because Glean doesn't know about it yet.  It is included in the first ping after being recorded (Ping 2), which causes it to be cleared.
+
+- **B. Ping lifetime, set on init and change**: The default value is included in Ping 1, and the changed value is included in Ping 2, which causes it to be cleared.  It therefore misses Ping 3, but when the application is started, it is recorded again and it is included in Ping 4.  However, this causes it to be cleared again and it is not in Ping 5.
+
+- **C. Application lifetime, set on change**: The value isn't included in Ping 1, because Glean doesn't know about it yet. After the value is changed, it is included in Pings 2 and 3, but then due to application restart it is cleared, so it is not included until the value is manually toggled again.
+
+- **D. Application, set on init and change**: The default value is included in Ping 1, and the changed value is included in Pings 2 and 3. Even though the application startup causes it to be cleared, it is set again, and all subsequent pings also have the value.
 
-Metric identifiers (the combination of a metric's category and name) must be unique across all metrics that are sent by an application.
+- **E. User, set on change**: The default value is missing from Ping 1, but since `user` lifetime metrics aren't cleared unless the user profile is reset, it is included in all subsequent pings.
+
+- **F. User, set on init and change**: Since `user` lifetime metrics aren't cleared unless the user profile is reset, it is included in all pings.
+
+Note that for all of the metric configurations, the toggle of the preference off and on during Ping 4 is completely missed.  If you need to create a ping containing one, and only one, value for this metric, consider using a [custom ping](pings/custom.html) to create a ping whose lifetime matches the lifetime of the value.
+
+#### What if none of these lifetimes are appropriate?
+
+If the timing at which the metric is sent in the ping needs to closely match the timing of the metrics value, the best option is to use a [custom ping](pings/custom.html) to manually control when pings are sent.
+
+### What should this new metric be called?
+
+#### Reuse names from other applications
+
+There's a lot of value using the same name for analogous metrics collected across different products. For example, BigQuery makes it simple to join columns with the same name across multiple tables. Therefore, we encourage you to investigate if a similar metric is already being collected by another product and use the same naming whenever feasible.
```

Comment:
Thinking again of this, if some metric is really shared across products of the same platform, maybe devs should consider writing a component/library using glean?

Reply:
I think it's a good idea to add a note about that, but it may not always be feasible.  Think of "open new tab", for example.  That's handled by a completely different code in Desktop vs. Fenix that it probably doesn't make sense to merge, even though they both have that "concept" of "open new tab".

I think maybe we say something like:

"Therefore, we encourage you to investigate if a similar metric is already being collected by another product.  If it is, there may be an opportunity for code reuse across these products.  But even if that's not feasible, use the same metric name for similar actions and concepts whenever possible."

Reply:
Good enough! Thank you for your patience!